src/core/lib/iomgr/tcp_server_posix.c

```diff
@@ -70,6 +71,8 @@
 
 #define MIN_SAFE_ACCEPT_QUEUE_SIZE 100
 
+int grpc_tcp_server_posix_expand_wildcard_addr = 0;
```

Comment:
Channel args could be set by ServerBuilder::SetOption(). A server builder plugin might not be needed.
In C level, a possible approach to set the default value is checking the key word for this flag in channel args. If it does not exist, configure this flag based on probing local hardware.